src/Rings/mpoly-local.jl

```diff
@@ -30,26 +30,46 @@ end
 struct MPolyElemLoc{T} <: AbstractAlgebra.RingElem where {T}
   frac::AbstractAlgebra.Generic.Frac
   parent::MPolyRingLoc{T}
-  function MPolyElemLoc(f::MPolyElem{T}, m::Oscar.MPolyIdeal) where {T}
-    R = parent(f)
-    (R != base_ring(m)) && error("Parent rings do not match!")
-    r = new{T}(f//R(1), Localization(R, m))
-    return r
-  end
-  function MPolyElemLoc(f::AbstractAlgebra.Generic.Frac, m::Oscar.MPolyIdeal)
-    R = parent(numerator(f))
+
+  # pass with checked = false to skip the non-trivial denominator check
+  function MPolyElemLoc{T}(f::AbstractAlgebra.Generic.Frac,
+                           p::MPolyRingLoc{T}, checked = true) where {T}
+    R = p.base_ring
     B = base_ring(R)
-    (R != base_ring(m)) && error("Parent rings do not match!")
-    pt = leading_coefficient.([gen(R, i)-m.gens.Ox[i] for i in 1:nvars(R)]) # This should be easier, somehow ...
-    (evaluate(denominator(f), pt) == base_ring(R)(0)) && error("Element does not belong to the localization.")
-    r = new{elem_type(B)}(f, Localization(R, m))
+    R != parent(numerator(f)) && error("Parent rings do not match")
+    T != elem_type(B) && error("Type mismatch")
+    if checked
+      # this code seems to assume m.gens is of the form [xi - ai]_i
+      m = p.max_ideal
+      # This should be easier, somehow ...
+      pt = leading_coefficient.([gen(R, i)-m.gens.Ox[i] for i in 1:nvars(R)])
+      if evaluate(denominator(f), pt) == base_ring(R)(0)
+        error("Element does not belong to the localization.")
+      end
+    end
+    return new{T}(f, p)
```

Comment:
I think this now also works (but of course it's also fine as-is)
```suggestion
    return new(f, p)
```